js/date-picker/js/lumx.date_picker_directive.js

```diff
@@ -72,6 +83,29 @@ angular.module('lumx.date-picker', [])
             generateCalendar();
         };
 
+        // selection of year
+        $scope.selectYear = function(year) {
+
+            // update selectedDate object
+            $scope.selectedDate.date.year(year);
+            // update model
+            $scope.model = $scope.selectedDate.date.toDate();
+            // update month based on current year
+            var year_diff = year - $scope.currentSelectedYear;
+            if(year_diff < 0) {
+                $scope.month = $scope.month.subtract(12 * Math.abs(year_diff), 'month');
```

Comment:
In which case does changing the month is required? I'm not sure it's necessary. But if you can give a reason why, I'll be happy to know it.